src/libAtomVM/term.c

```diff
@@ -499,14 +513,52 @@ TermCompareResult term_compare(term t, term other, TermCompareOpts opts, GlobalC
             break;
 
         } else if (term_is_reference(t) && term_is_reference(other)) {
```

Comment:
Let's be coherent with what we are doing with pids, and let's make 2 branches for local and external refs instead of nesting them inside term_is_reference.

Reply:
Actually external pids/refs and local pids/refs comparison is a little bit more complex in BEAM and this section was rewordked.